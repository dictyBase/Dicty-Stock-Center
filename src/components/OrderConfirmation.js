// @flow
import React from "react"
import { Link } from "react-router-dom"
import Grid from "@material-ui/core/Grid"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { AlertSuccess, PrimaryLargeButton } from "styles"

type Props = {
  /** the current order ID number */
  id: number,
}

/**
 * Displays notification that the user's order was submitted successfully.
 */

export const OrderConfirmation = (props: Props) => (
  <Grid container wrap="wrap" justify="center">
    <Grid item xs={8}>
      <Grid container wrap="wrap" justify="center">
        <Grid item xs={12}>
          <AlertSuccess>
            <FontAwesomeIcon icon="check-circle" size="5x" />
            <h3>Thank you, your order has been submitted successfully!</h3>
            <p>
              Order Number: <strong>{props.id}</strong>
            </p>
            <p>We have sent you a confirmation email.</p>
            <p>
              The <strong>Payer</strong> will soon receive emails through the{" "}
              <strong>NU Core</strong> system to complete payment.
            </p>
          </AlertSuccess>
        </Grid>
        <Grid item xs={12}>
          <PrimaryLargeButton>
            <Link to="/">
              <FontAwesomeIcon icon="home" /> Stock Center Home
            </Link>
          </PrimaryLargeButton>
        </Grid>
      </Grid>
    </Grid>
  </Grid>
)

export default OrderConfirmation
