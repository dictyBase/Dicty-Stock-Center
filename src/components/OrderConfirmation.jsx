import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { Flex, Box } from 'rebass'
import FontAwesome from 'react-fontawesome'
import { AlertSuccess } from 'styles'
import { PrimaryButton } from 'styles/buttons'

const CenteredBox = styled(Box)`
    text-align: center;
`

export default class OrderConfirmation extends Component {
    displayName = 'order confirmation'

    static propTypes = {
        order: PropTypes.object
    }

    render() {
        const { id } = this.props.order
        return (
            <Flex wrap justify="center">
                <Box w={ '75%' }>
                    <Flex wrap justify="center">
                        <CenteredBox w={ 1 }>
                            <AlertSuccess>
                                <FontAwesome name="check-circle-o" size="5x" />
                                <h3>
                                    Thank you, your order has been submitted
                                    successfully!
                                </h3>
                                <p>
                                    Order Number: <strong>{ id }</strong>
                                </p>
                                <p>We have sent you a confirmation email.</p>
                                <p>
                                    The <strong>Payer</strong> will soon receive
                                    emails through the <strong>NU Core</strong>{ ' ' }
                                    system to complete payment.
                                </p>
                            </AlertSuccess>
                        </CenteredBox>
                        <Box w={ 1 }>
                        <PrimaryButton large block>
                            <Link to="/">
                                <FontAwesome name="home" /> Stock Center Home
                            </Link>
                            </PrimaryButton>
                        </Box>
                    </Flex>
                </Box>
            </Flex>
        )
    }
}
