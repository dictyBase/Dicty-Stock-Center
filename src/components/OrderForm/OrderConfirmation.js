// @flow
import React from "react"
import { Link } from "react-router-dom"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import Button from "@material-ui/core/Button"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"

const styles = theme => ({
  alertBox: {
    padding: "15px",
    marginBottom: "20px",
    border: "1px solid transparent",
    borderRadius: "4px",
    color: "#3c763d",
    backgroundColor: "#dff0d8",
    borderColor: "#d6e9c6",
    textAlign: "center",
  },
  btn: {
    marginBottom: "10px",
    backgroundColor: "#004080",
    "&:hover": {
      backgroundColor: "#0073e6",
    },
  },
})

type Props = {
  /** Material-UI styling */
  classes: Object,
}

/**
 * Displays notification that the user's order was submitted successfully.
 */

export const OrderConfirmation = (props: Props) => {
  const { classes } = props

  return (
    <Grid container wrap="wrap" justify="center">
      <Grid item xs={8}>
        <Grid container wrap="wrap" justify="center">
          <Grid item xs={12}>
            <div className={classes.alertBox}>
              <FontAwesomeIcon icon="check-circle" size="5x" />
              <p>We have sent you a confirmation email.</p>
              <p>
                The <strong>Payer</strong> will soon receive emails through the{" "}
                <strong>NU Core</strong> (Northwestern University) system to
                complete payment.
              </p>
            </div>
          </Grid>
          <Grid item>
            <Button
              component={Link}
              to="/"
              color="primary"
              variant="contained"
              size="large"
              className={classes.btn}>
              <FontAwesomeIcon icon="home" /> &nbsp; Stock Center Home
            </Button>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}

export default withStyles(styles)(OrderConfirmation)
