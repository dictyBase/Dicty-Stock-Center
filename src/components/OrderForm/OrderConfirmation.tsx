// @flow
import React from "react"
import { Link } from "react-router-dom"
import Grid from "@material-ui/core/Grid"
import Button from "@material-ui/core/Button"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import useStyles from "./formStyles"

/**
 * Displays notification that the user's order was submitted successfully.
 */

export const OrderConfirmation = () => {
  const classes = useStyles()

  return (
    <Grid container wrap="wrap" justify="center">
      <Grid item xs={8}>
        <Grid container wrap="wrap" justify="center">
          <Grid item xs={12}>
            <div className={classes.alertBox}>
              <h1>Success!</h1>
              <FontAwesomeIcon icon="check-circle" size="5x" />
              <p>We have sent you a confirmation email.</p>
              <p>
                The <strong>Payer</strong> will soon receive emails through the{" "}
                <strong>NU Core</strong> (Northwestern University) system to
                complete payment.
              </p>
            </div>
          </Grid>
          <Grid item>
            <Button
              component={Link}
              to="/"
              color="primary"
              variant="contained"
              size="large"
              className={classes.btn}>
              <FontAwesomeIcon icon="home" /> &nbsp; Stock Center Home
            </Button>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}

export default OrderConfirmation
