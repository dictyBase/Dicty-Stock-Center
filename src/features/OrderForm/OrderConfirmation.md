```js
const location = {
  state: {
    orderID: "123456",
  },
}
;<OrderConfirmation location={location} />
```
