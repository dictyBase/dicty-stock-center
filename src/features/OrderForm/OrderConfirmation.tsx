import React from "react"
import { Link, Redirect } from "react-router-dom"
import { makeStyles, Theme } from "@material-ui/core/styles"
import Box from "@material-ui/core/Box"
import Grid from "@material-ui/core/Grid"
import Button from "@material-ui/core/Button"
import Typography from "@material-ui/core/Typography"
import { grey } from "@material-ui/core/colors"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { PDFViewer } from "@react-pdf/renderer"
import OrderSummaryPDF from "./Submit/OrderSummaryPDF"
import { CartItem } from "common/types"
import { FormikValues } from "./utils/initialValues"

const useStyles = makeStyles((theme: Theme) => ({
  container: {
    backgroundColor: grey[200],
    margin: theme.spacing(2, 0, 2, 0),
    minHeight: "320px",
  },
  button: {
    "&:hover": {
      color: "#fff",
    },
  },
  confirmation: {
    fontSize: "1.2em",
  },
}))

type LocationProps = {
  location: {
    /** URL Pathname */
    pathname: string
    /** State passed as props from history.push */
    state?: {
      /** Submitted order ID */
      orderID: string
      /** Object containing all entered form data */
      formData: FormikValues
      /** All items from this order */
      cartItems: CartItem[]
      /** Total cost of items in cart */
      cartTotal: string
    }
  }
}

/**
 * Displays notification that the user's order was submitted successfully.
 */

const OrderConfirmation = ({ location }: LocationProps) => {
  const classes = useStyles()
  const { state } = location

  if (state === undefined) {
    return <Redirect to="/" />
  }

  return (
    <Grid container alignItems="center" className={classes.container}>
      <Grid item xs={12}>
        <Box margin={2}>
          <Typography gutterBottom variant="h1">
            <FontAwesomeIcon icon="check-circle" /> Thank you for your order
          </Typography>
          <Box mt={3} mb={3}>
            <Typography
              gutterBottom
              component="p"
              className={classes.confirmation}>
              <strong>Order ID: {state.orderID}</strong>
            </Typography>
          </Box>
          <Box mb={3}>
            <Typography gutterBottom component="p">
              We have sent you a confirmation email.
            </Typography>
            <Typography gutterBottom component="p">
              The <strong>Payer</strong> will soon receive emails through the{" "}
              <strong>NU Core</strong> (Northwestern University) system to
              complete payment.
            </Typography>
          </Box>
          <PDFViewer width={window.innerHeight} height={500}>
            <OrderSummaryPDF
              cartItems={state.cartItems}
              formData={state.formData}
              cartTotal={state.cartTotal}
              orderID={state.orderID}
            />
          </PDFViewer>
          <Button
            component={Link}
            to="/"
            color="primary"
            variant="contained"
            size="large"
            className={classes.button}
            startIcon={<FontAwesomeIcon icon="home" />}>
            DSC Home
          </Button>
        </Box>
      </Grid>
    </Grid>
  )
}

export default OrderConfirmation
