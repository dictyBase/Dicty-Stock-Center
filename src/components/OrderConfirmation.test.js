import React from "react"
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { OrderConfirmation } from "./OrderConfirmation"

test("matching a snapshot of OrderConfirmation", () => {
  const order = {
    id: "test"
  }

  const component = renderer.create(
    <BrowserRouter>
      <OrderConfirmation order={order} />
    </BrowserRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
