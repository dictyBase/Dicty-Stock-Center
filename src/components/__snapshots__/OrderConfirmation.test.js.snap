// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of OrderConfirmation 1`] = `
.c2 {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0;
  font-size: 14px;
  font-weight: normal;
  line-height: 1.42857143;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-image: none;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #fff;
  background-color: #337ab7;
  border-color: #2e6da4;
  display: block;
  width: 100%;
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
}

.c2.block {
  display: block;
  width: 100%;
}

.c2.large {
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
}

.c2.xs {
  padding: 1px 5px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 3px;
  width: 20%;
}

.c2 a {
  color: #fff;
}

.c2 a:hover {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:focus {
  color: #fff;
  background-color: #286090;
  border-color: #122b40;
}

.c2:hover {
  color: #fff;
  background-color: #286090;
  border-color: #204d74;
}

.c2:active {
  color: #fff;
  background-color: #286090;
  border-color: #204d74;
}

.c1 {
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #3c763d;
  background-color: #dff0d8;
  border-color: #d6e9c6;
  text-align: center;
}

.c0 {
  text-align: center;
}

<div
  className="sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="csVOHI"
    is={null}
  >
    <div
      className="sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="c0 gRrvFh"
        is={null}
      >
        <div
          className="c1"
        >
          <span
            aria-hidden={true}
            className="fa fa-check-circle-o fa-5x"
          />
          <h3>
            Thank you, your order has been submitted successfully!
          </h3>
          <p>
            Order Number: 
            <strong />
          </p>
          <p>
            We have sent you a confirmation email.
          </p>
          <p>
            The 
            <strong>
              Payer
            </strong>
             will soon receive emails through the 
            <strong>
              NU Core
            </strong>
             system to complete payment.
          </p>
        </div>
      </div>
      <div
        className="gRrvFh"
        is={null}
      >
        <button
          className="c2"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <span
              aria-hidden={true}
              className="fa fa-home"
            />
             Stock Center Home
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
